### ISSUE TITLE
Force React to hit server route instead of staying within the SPA

### ISSUE THREAD

**Issue by greeddiscient** [facebook/react#10034]
Hello,

I've been facing this problem for quite some time and none of the solutions I found through google worked. Basically what I'm trying to do is make my React App go out of the SPA and hit my server route on Express (e.g. app.get(/auth/steam)

However I've tried using react-router's Link and Redirect and it still stayed within the SPA and then I found out react-router is only for internal navigation. I tried using a the anchor tag which supposedly forces the app to hit the server route but it still stays within the React page and gives me my 404 component. 

The weird thing is that when I hard refresh (cmd shift R on mac) when I'm on localhost:3000/auth/steam it hits the server route and redirects to the other website's login page (in my case Steam login). After I finish logging in the callback url is /auth/steam/return which it shows again as the 404 component, but when I hard refresh again it does everything it's supposed to and redirects to my homepage because I set the redirect to / and I also have the req.user available.

I know I gave the example of what I'm trying to do (user authentication) but I think this question is not specific to what I'm trying to do but it's more of a React question because if I wanted to use another server route to do something else (app.get(exampleURL) I would face the same problem.

So basically my question is how do I force React to hit the server route instead of staying within the SPA

**Comment by jack118104** [comment]
you can use <a href="your server path"></a>,just easy

**Comment by greeddiscient** [comment]
@jack118104 is that an incomplete answer?

**Comment by aweary** [comment]
We try to use this issue tracker for bug reports and feature requests only. If you have a usage or support question, we recommend checking out one of the great community-driven platforms like [Reactiflux](https://www.reactiflux.com/), [discuss.reactjs.org](https://discuss.reactjs.org/), or StackOverflow.
